src/foam/nanos/medusa/MedusaConsensusMonitor.js

```diff
@@ -168,11 +179,16 @@ foam.CLASS({
         alarm.setNote("No Consensus: "+next.toSummary());
         alarm = (Alarm) alarmDAO.put(alarm);
 
+        // TOOD: getMinReplayInterval has to be adjusted based on count.
+        // Nodes with multiple mediator requests are constrained on memory and
+        // 1m records might take 10 minutes to read and send.
         if ( getLastReplay() == 0L ||
              System.currentTimeMillis() - getLastReplay() >= getMinReplayInterval() ) {
           logger.info("request replay");
           final ReplayRequestCmd cmd = new ReplayRequestCmd();
-          cmd.setDetails(new ReplayDetailsCmd.Builder(x).setMinIndex(next.getIndex()).build());
+          // request parents as well to handle 'parent not found' scenario
+          Long min = Math.min(next.getIndex(), Math.min(next.getIndex1(), next.getIndex2()));
```

Comment:
Why Long instead of long?

Reply:
I'm continually switching between the two - with casting count, max, ... sinks.  
changed to long.